ecolex/legislation.py

```diff
@@ -190,20 +178,32 @@ def harvest_file(upfile):
                 legislation['legCountry_es'] = fao_country.get('es')
                 legislation['legCountry_fr'] = fao_country.get('fr')
 
-        legYear = legislation.get('legYear')
-        if legYear:
+                region = json_regions.get(fao_country.get('en'))
+                if region:
+                    legislation['legGeoArea_en'] = region.get('en')
+                    legislation['legGeoArea_fr'] = region.get('fr')
+                    legislation['legGeoArea_es'] = region.get('es')
+
+        if 'legDate' in legislation:
             try:
-                legDate = datetime.strptime(legYear, '%Y')
+                legDate = datetime.strptime(legislation.get('legDate'), '%Y-%m-%d')
+                legislation['legYear'] = legDate.strftime('%Y')
                 legislation['legDate'] = legDate.strftime('%Y-%m-%dT%H:%M:%SZ')
             except Exception:
-                logger.debug('Error parsing legYear %s' % (legYear))
-
-        url_value = document.attrs.get('url', None)
-        if url_value:
-            legislation['legLinkToFullText'] = url_value
+                logger.debug(f"Error parsing legDate {legislation['legDate']}")
+
+        filenames = get_content(document.findAll('link_to_full_text'))
+        url_values = []
+        for filename in filenames:
+            extension = filename.rsplit('.')[-1]
```

Comment:
You can transform it to lowercase here, since you're only using the lowercase version.